api/comments/serializers.py

```diff
@@ -62,7 +62,7 @@ def get_is_abuse(self, obj):
         user = self.context['request'].user
         if user.is_anonymous():
             return False
-        return user._id in obj.reports
+        return user._id in obj.reports and not obj.reports[user._id]['retracted']
```

Comment:
Is there a chance that `'retracted'` could cause a `KeyError`? Would you rather use `.get()` with a default or wrap this and the one on line 190 in a `try`? If it's just broken beyond repair if there is a `KeyError` then it's probably worth wrapping it in a try and responding to the user in a more gentle way than a 500.